### PR TITLE
BLD: Move config.h creation to `setup` stage

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @HaoZeke,

I decided to spend some time to actually move `config.h` creation to `setup` stage (with existing `getarch`, etc.) anyway. It was too annoying to work with manually created files.

This refactor is separate from your `config.h` work. It's a rework of my previous `config.h` generation attempt, and it relies on `run_command`s that are executed in the `setup` stage.

### CODE REVIEW

File: write_to_file.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import argparse
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Write to file.")
+    parser.add_argument("contents", help="Contents.")
+
+    args = parser.parse_args()
+
+    f = open("./build/config.h", "a")
```
Comment (HaoZeke) [inline write_to_file.py]: This (build directory) needs to be passed to the script as an argument

File: meson.build [hunk 2]
```diff
@@ -462,55 +462,23 @@ symb_defs = {
...
+run_command('./f_check', 'Makefile.conf', _config_h, fc_id, check: true)
+
+run_command(cc_id, '-o', 'getarch', 'getarch.c', 'cpuid.S', check: true)
+_getarch_result = run_command('./getarch', '1', check: true, capture: true)
+run_command(py3, './write_to_file.py', _getarch_result.stdout(), check: true)
+
+run_command(cc_id,
+            '-DGEMM_MULTITHREAD_THRESHOLD=4',
+            '-I.',
+            '-I./build',
```
Comment (HaoZeke) [inline meson.build]: Should be project build root I think?

**Review by HaoZeke — CHANGES_REQUESTED** [review]

**Comment by HaoZeke** [comment]
Looks much cleaner, are the run commands guaranteed to be executed in order in the configuration stage?

**Comment by mtsokol** [comment]
> Looks much cleaner, are the run commands guaranteed to be executed in order in the configuration stage?

In my opinion - yes. They're always executed line by line.

File: write_to_file.py (hunk 1 above)
Comment (mtsokol) [reply write_to_file.py]: Done!

File: meson.build (hunk 2 above)
Comment (mtsokol) [reply meson.build]: Yes - done!

**Review by HaoZeke — APPROVED** [review]

**Comment by HaoZeke** [comment]
Thanks @mtsokol

**Pull request merged**